{% load static %}

<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>local sequence</title>
    <link rel="stylesheet" type="text/css" href=" {% static '/css/localalignment.css' %}">
</head>
<body>
    <nav class="logo ">
        
        <h1>algo4bio</h1>
</nav>
    <header>
        <div class="main">
            <h1>Local Sequence</h1>
    </div>

   </header> 
   <br>
   <section>
    <div class="seq">

     <form method="POST" enctype="multipart/form-data">
        {% csrf_token %}
        <label><h3>Please enter the first sequence</h3> </label>
        <input id="text" type="text" name="seq1" placeholder="First sequence" />
        <br>
        <br>
        <input type="file" id="myfile" name="document1">
        <br>
        <label> <h3>Please enter the second sequence</h3></label>
        <input id="text" type="text" name = "seq2"placeholder="Second sequence"/>
        <br>
        <br>
        <input type="file" id="myfile" name="document2">
        <br>
        <br>
        <br>

            
        <input class="text" type="text" name="match" placeholder="match"/>
        <input class="text" type="text" name="mismatch" placeholder="mismatch"/>   
        <input class="text" type="text" name="gap" placeholder="gap"/>
        <br>
        <br>
        <br>
        <br>
       
        <button type="submit" id="button" name="calc" value="-1">Enter</button>
        <br>
        
        <input id="reset" type="reset"/>
        <br>
        
        <button type="submit" id="button" name="back" value="-1">use another algorithm</button>
       
      
    </form>

    </div>
    <hr>
    <hr>
    <div class="txt">
        <h2>Working of Smith-Waterman Algorithm :</h2>
        <br>
        <h3>Intialization of Matrix </h3>
        <p>
            The basic steps for the algorithm are similar to that of Needleman-Wunsch algorithm. The steps are:
            <br>
    
    1. Initialization of a matrix.
    2. Matrix Filling with the appropriate scores.
    3. Trace back the sequences for a suitable alignment.
    <br>
    <br>
    To study the Local sequence alignment consider the given below sequences.
    <br>
    <br>
    CGTGAATTCAT (sequence#1 or A)
    GACTTAC (sequence #2 or B)
    
     <br>
     <br>
    
    The two sequences are arranged in a matrix form with A+1columns and B+1rows. The values in the first row and first column are set to zero as shown in Figure 1.
    
    <figure>
        <img src="{% static 'images/local alignment/Figure 1(3).jpg' %}" alt="">
        <figcaption>Figure 1: Initialization of Matrix</figcaption>
    </figure>
    <br>
    Variables used:
    <br>
              i,j describes row and columns.
              <br>
              M is the matrix value of the required cell (stated as Mi,j)
              <br>
              S is the score of the required cell (Si, j)
              <br>
              W is the gap alignment
     
        </p>
        <h3>Matrix Filling</h3>
        <p>
            The second and crucial step of the algorithm is filling the entire matrix, so it is more important to know the neighbor values (diagonal, upper and left) of the current cell to fill each and every cell.
            <br>
            <img src="{% static 'images/local alignment/showimage.png' %}" alt="">
            <br>
            As per the assumptions stated earlier, fill the entire matrix using the assumed scoring schema and initial values.  One can fill the 1st row and 1st column with the scoring matrix as follows.
    <br>
    <br>
    
     
    
    The first residue (nucleotides or amino acids) in both sequences is ‘C’ and ‘G’, the matching score or the mismatching score is going to be added the neighboring value which is diagonally located i.e. 0. The upper and left values are added to the gap penalty score from the matrix. So the scoring schema equation can be shown as follows.
    <br>
    <br>
    <img src="{% static 'images/local alignment/showimage (1).png' %}" alt="">

    <br>
    From the above calculations the maximum value obtained is 0. Finding the maximum value for Mi,j position, one can notice that there is no chance to see any negative values in the matrix, since we are taking 0 as lowest value.
    
    <br>
    <br>
     
    
    After filling the matrix, keep the pointer back to the cell from where the maximum score has been determined. In the similar fashion fill all the values of the matrix of the cell.
    
    <br>
    <br>
      
    
    For the example the matrix can be filled is shown in Figure 2.
    <figure>
        <img src="{% static 'images/local alignment/Figure 2(3).jpg' %}" alt="">
        <figcaption>Figure 2: Matrix filling with back pointers</figcaption>
        
    </figure>
    Each cell is back pointed by one or more pointers from where the maximum score has been obtained.
     
        </p>
        <h3>Trace backing the sequences for an optimal alignment:</h3>
        <p>
            The final step for the appropriate alignment is trace backing, prior to that one needs to find out the maximum score obtained in the entire matrix for the local alignment of the sequences.It is possible that the maximum scores can be present in more than one cell, in that case there may be possibility of two or more alignments, and the best alignment by scoring it.
    
            <br>
            <br>
    
    In this example we can see the maximum score in the matrix as 18, which is found in two positions that lead to multiple alignments, so the best alignment has to be found.
    
    <br>
    <br>
    
    So the trace back begins from the position which has the highest value, pointing back with the pointers,  thus find out the possible predecessor, then move to next predecessor and continue until we reach the score 0 (Figure 3).
    
    <figure>
        <img src="{% static 'images/local alignment/Figure 3(3).jpg' %}" alt="">

        <figcaption>Figure 3: Trace back of first possible alignment</figcaption>
    </figure>
    It is possible to find two pointers pointing out from one cell, where both ways(alignments) can be considered,  best one is found by scoring and finding maximum score among them.
    <figure>
        <img src="{% static 'images/local alignment/Figure 4(2).jpg' %}" alt="">

        <figcaption>Figure 4: Trace back of second possible alignment</figcaption>
    </figure>
    Thus a local alignment is obtained and one can see the possible alignments as in Figure 5.
    <figure>
        <img src="{% static 'images/local alignment/Figure 5(2).jpg' %}" alt="">

        <figcaption>Figure 5: Scoring for best alignment</figcaption>
    
    </figure>
    The two alignments can be given with a score, for matching as +5 , mismatch as -3 and gap penalty as -4, sum up all the individual scores and the alignment which has maximum score after this can be taken as the best alignment.
    
     <br>
     <br>
    
    By summing up the scores both of the alignments are giving the same as 18, so one can predict both alignments are the best.
        </p>
    </div>
    
   
</body>
</html>